src/apps/external_executor/grpc.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#include "ccf/odata_error.h"
+#include "ds/serialized.h"
+
+#include <arpa/inet.h>
+#include <variant>
+#include <vector>
+
+namespace grpc
+{
+  using CompressedFlag = uint8_t;
+  using MessageLength = uint32_t;
+
+  static constexpr size_t message_frame_length =
+    sizeof(CompressedFlag) + sizeof(MessageLength);
+
+  MessageLength read_message_frame(const uint8_t*& data, size_t& size)
+  {
+    auto compressed_flag = serialized::read<CompressedFlag>(data, size);
+    if (compressed_flag > 1)
+    {
+      throw std::logic_error(fmt::format(
+        "gRPC compressed flag has unexpected value {}", compressed_flag));
+    }
+    return ntohl(serialized::read<MessageLength>(data, size));
```

Comment:
We're not storing the `compressed_flag` anywhere, so I think we're actually expecting that it's always `0`? If it is ever `1`, then the contents are encoded based on the `Message-Encoding` header, and I don't think we handle that case anywhere? Fine if we don't for now, but we should make this error more precise (`>1` => unexpected value, `!=0` => unhandled value).

Reply:
The intent was to throw if the value is unexpected or compression is enabled but I clearly can't write a comparison operator.  Fixed now (`>= 1`).